test-case/latency-metrics.sh

```diff
@@ -0,0 +1,287 @@
+
+OPT_NAME['S']='shorts'          OPT_DESC['S']='Try to use 16-bit samples instead of 32-bit, if possible.'
+OPT_HAS_ARG['S']=0              OPT_VAL['S']=0
+
+OPT_NAME['P']='port_p'          OPT_DESC['P']='JACK playback port with loopback.'
+OPT_HAS_ARG['P']=1              OPT_VAL['P']='system:playback_1'
+
+OPT_NAME['C']='port_c'          OPT_DESC['C']='JACK capture port with loopback.'
+OPT_HAS_ARG['C']=1              OPT_VAL['C']='system:capture_1'
+
+OPT_NAME['s']='sof-logger'      OPT_DESC['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_HAS_ARG['s']=0              OPT_VAL['s']=1
+
+OPT_NAME['v']='verbose'         OPT_DESC['v']='Verbose logging.'
+OPT_HAS_ARG['v']=0              OPT_VAL['v']=0
+
+OPT_NAME['X']='xruns-ignore'    OPT_DESC['X']="How many 'xrun' errors to ignore."
+OPT_HAS_ARG['X']=1              OPT_VAL['X']=0
+
+OPT_NAME['L']='loopback'        OPT_DESC['L']="Set internal loopback at JACK instead of ports."
+OPT_HAS_ARG['L']=0              OPT_VAL['L']=0
+
+OPT_NAME['t']='trial'           OPT_DESC['t']="Trial mode: repeat measurements decreasing buffer."
+OPT_HAS_ARG['t']=0              OPT_VAL['t']=0
+
+OPT_NAME['T']='trial-until'     OPT_DESC['T']="Trial mode: repeat until this value."
+OPT_HAS_ARG['T']=1              OPT_VAL['T']=2
+
+func_opt_parse_option "$@"
+
+alsa_device=${OPT_VAL['d']}
+pcm_p=${OPT_VAL['p']}
+pcm_c=${OPT_VAL['c']}
+alsa_shorts=$([ "${OPT_VAL['S']}" -eq 1 ] && echo '--shorts' || echo '')
+port_playback=${OPT_VAL['P']}
+port_capture=${OPT_VAL['C']}
+rate=${OPT_VAL['r']}
+run_period=${OPT_VAL['R']}
+run_verbose=$([ "${OPT_VAL['v']}" -eq 1 ] && echo '--verbose' || echo '')
+max_xruns=${OPT_VAL['X']}
+set_loopback=${OPT_VAL['L']}
+jackd_frames=${OPT_VAL['f']}
+jackd_period=${jackd_frames}
+jackd_nperiods=${OPT_VAL['n']}
+trial_mode=${OPT_VAL['t']}
+trial_until=${OPT_VAL['T']}
+
+next_sep=""
+
+METRICS_JSON="${LOG_ROOT}/metrics_iodelay.json"
+EVENTS_JSON="${LOG_ROOT}/events_jackd.json"
+RESULT_JSON="${LOG_ROOT}/test_result.json"
+RUN_PERIOD_MAX="$((run_period + 30))s"
+JACKD_TIMEOUT="$((run_period + 15))s"
+JACKD_OPTIONS="${run_verbose} --realtime --temporary"
+JACKD_BACKEND="alsa"
+JACKD_BACKEND_OPTIONS="-n ${jackd_nperiods} -r ${rate} ${alsa_shorts}"
+WAIT_JACKD="2s"
+IODELAY_TIMEOUT="${run_period}s"
+WAIT_IODELAY="2s"
+
+if [ -n "${alsa_device}" ]; then
+  if [ -n "${pcm_p}" ] || [ -n "${pcm_c}" ]; then
+    skip_test "Give either ALSA device, or ALSA playback/capture pcm-s."
+  fi
+  JACKD_BACKEND_OPTIONS="-d ${alsa_device} ${JACKD_BACKEND_OPTIONS}"
+elif [ -z "${pcm_p}" ] || [ -z "${pcm_c}" ]; then
+    skip_test "No playback or capture ALSA PCM is specified."
+else
+  JACKD_BACKEND_OPTIONS="-P ${pcm_p} -C ${pcm_c} ${JACKD_BACKEND_OPTIONS}"
+fi
+
+if [ "${set_loopback}" == 1 ]; then
+  port_playback="${set_loopback}"
+  port_capture="${set_loopback}"
+fi
+
+if [ -z "${port_playback}" ] || [ -z "${port_capture}" ]; then
+  skip_test "No playback or capture JACK port is specified."
```

Comment:
Most/all the above should be in the `main` function (#740), which can/should be moved here.